dc backgrounds and job post (#580)
deltachat/deltachat-pages: _posts/2022-12-13-decentuidevs.md

---

> title: Decentralization Unchained (Psst ... want a UI/UX dev job?)

Reviewer: ```suggestion
title: Decentralization Unchained + UI/UX dev job
```

author: olgax
image: ../assets/blog/decentralization-unchained.jpg
---

What do indigenous communities in the Amazon rainforest, friends in Cuba, families in Iran, activists from Russia, a monastery in Cambodia, and many folks in the Fediverse have in common?

![Artwork Decentralization Unchained](../assets/blog/decentralization-unchained.jpg)

You guessed it? They're using or following Delta Chat because it is:

> - **adaptable**: Indigenous communities

Reviewer 2: i'd use the first letter uppercase here and as well as for  the other three points.

```suggestion
- **Adaptable**: Indigenous communities
```

Since the beginnings of Delta Chat in a small city north of Hamburg almost five years ago,
we have posted little about the wonderful and diverse ways people use our apps.
We usually prefer to share completed releases instead of pre-announcing or
trying to create hype by boasting about users and growth.
Avoiding popularity is not merely born out of humility, but also realism as
we can only address so much feedback and suggestions.
If we are over capacity, feature requests and ideas turn into pressure and stress.
This post is not a pre-announcement but a ride through what's currently developing,
about how we are doing things and about job offers to join our UX-oriented developments.

### Sharing web apps in a chat (webxdc)

<img src="../assets/blog/2022-12-webxdc-poll.jpg" width="270" style="float:right; margin-left:1em;" />
[Sharing web apps in a chat (webxdc)](https://webxdc.org) was released in mid-2022.
Dozens of grassroots-ported games are now played everyday in chat groups and through mailing lists.
Collaborative tooling apps (polls, checklists, calendars, editors) are evolving.
No permission from us, or any app store, is needed to create and distribute your apps.

> Programming newcomers succeeded in delivering working usable apps via a zip file

Reviewer 2: ```suggestion
Programming newcomers succeeded in delivering working usable apps via a ZIP file
```

> containing an `index.html`, some assets and JS/Typescript/CSS as preferred.

Reviewer 2: ```suggestion
containing an `index.html`, some assets and JS/TypeScript/CSS as preferred.
```

as with all webxdc apps shared in a chat,
it features these wonderful properties and UX conveniences:

- **end-to-end encrypted**

- **offline-first**,

- **without logins or GDPR/consent screens**.

<img src="../assets/blog/gordon_superapp.jpg" width="270" style="float:right; margin-left:1em;" />
UX and privacy researchers and hackers from diverse projects such as Cryptpad, DAT, IPFS,
Peermaps and Libreoffice agree:
Web tech combined with decentralized chat (aka "webxdc") is a rare jewel worth refining.

### Movements in the woods and underground…

<img src="../assets/blog/2022-12-stickers-and-beer.jpg" width="270" style="float:right; margin-left:1em;" />
With a wide variety of users, developers, partners, and co-operative projects
we discuss many wants and needs:
Supporting emoji-reactions, having Telegram-style larger channels,
evolving WhatsApp-style communities,
encryption improvements, seamless multi-device setup,
and even the holy grails of Peer-to-Peer messaging.
The somewhat incredible thing is that we are actually progressing
around several of these topics.
We just don't like to pre-announce much.

> This post here is a bit of an exception to prove the rule 🙂.

Reviewer 2: having the punctuation after the emoji does not look good and is usually not done, see eg. https://writersafterdark.net/2020/02/13/how-to-punctuate-emojis/

i'd go for
```suggestion
This post here is a bit of an exception to prove the rule 🙂
```
as done for 🤯 atop.

Reviewer: I didn't know about this, but it makes sense. My reflex is to put it after punctuation so that screen-readers end the sentence with the correct tone.

"Usable security" itself remains an evasive concept
if we can't facilitate its implementation in real-life apps.
Between usability, security, and implementation considerations
there is no natural hierarchy: each constrains and influences the others.
In 2023, in collaboration with [TeamUSEC](https://teamusec.de/) we want
to perform systematic user and field-testing around security topics,
in part following up on the often-quoted usable security study ["When Signal hits the fan"](https://dx.doi.org/10.14722/eurousec.2016.23012).

### Never underestimate the importance of testing

Luckily, we have a wonderful testing community with key contributors
sorting out issues with users and discussing with developers.
One from Austria kept reminding us about specific connectivity problems
even if they rarely occur and are hard to re-produce:
the issues were fixed the last week, which increased resiliency for everyone.
Did you know that most Delta Chat features are first tested on a Caribbean island?
Some were originally developed in the DeltaLab app:
a friendly fork of Delta Chat Android only available in the Cuban app store.

### Ways out and around the dreaded onboarding problem

> Using a pre-existing email account is one of Delta Chat's core features, but also

Reviewer 2: ```suggestion
Using a pre-existing e-mail account is one of Delta Chat's core features, but also
```

bots based on Rust-core are being deployed for practical and entertaining contexts.
But this post is getting long and twisted enough already so more on released bots some other time …

### We are looking for experienced React/Web, Java/Android and Swift/iOS developers

![Desktop background image screenshot](../assets/blog/screenshots/2020-01-15-delta-chat-desktop-client-release-background-image.jpg)
Our [Desktop app](https://github.com/deltachat/deltachat-desktop) uses Web technology (React, TypeScript), currently via
Electron, but maybe someday via Tauri so there's no need to ship a full browser with the app.
The desktop app is regularly released and published
to Windows, MacOS, and Linux distribution channels.
There are many things to improve regarding platform integration,
bug fixing and bringing about new or refined UI/UX interactions.

Our [Android app](https://github.com/deltachat/deltachat-android) is a 2019-fork of Signal's Java app,
diverged in some areas because we orient ourselves
around the interfaces of Telegram and WhatsApp.
The Android app remains our "flagship" app that is available in various stores,
among them Google Play and F-Droid.

> Our [iOS app](https://github.com/deltachat/deltachat-ios) is a self-development using Uikit and Swift bindings to core.

Reviewer 2: ```suggestion
Our [iOS app](https://github.com/deltachat/deltachat-ios) is a self-development using UIKit and Swift bindings to core.
```

If we gather somewhere remote then usually for weeks rather than days,
and without seeking much publicity.
We try to avoid constant busyness and urgency as it prevents enjoyable collaboration.
We aim to arrange ourselves [empathically](https://delta.chat/en/community-standards)
and keep dates and clocktime scheduling to a minimum.

Currently, there are around a dozen weekly committers to our repositories,
with a bit more than half of them receiving funding either
through employment contracts (if in Germany) or freelancing (if international).
Another dozen people and maintainers from other projects
are involved in background discussions and hack sessions around "what's interesting to do next, maybe together".
We work with people located on several continents, some of them exiled,
some of them migrants of choice, some of them in more stable and resource-rich environments.

### Please help us to further improve UX and UIs for everyone

Please contact us via e-mail addressed to delta@merlinux.eu
if you are interested in helping us and our users with beautiful front end developments.
Please provide some background on your prior projects or apps.

> A CV is not neccessary but welcome if you have one around.

Reviewer 2: ```suggestion
A CV is not necessary but welcome if you have one around.
```
